recipe/meta.yaml

```diff
@@ -31,18 +31,30 @@ source:
     folder: testing
 
 build:
-  number: 6
+  number: 7
   # for cuda support, building with one version is enough to be compatible with
   # all later versions, since arrow is only using libcuda, and not libcudart.
   skip: true  # [cuda_compiler_version not in ("None", cuda_compiler_version_min)]
   # arrow promises API- & ABI-compatibility along SemVer, see #1096
 
 requirements:
   build:
-    - {{ compiler("c") }}
     - {{ stdlib("c") }}
+    {% if cuda_compiler_version == "None" %}
```

Comment:
Have grouped the non-CUDA cases together as it seems those are the ones that are mainly being pinned